src/coreclr/tools/SuperFileCheck/Program.cs

```diff
@@ -80,6 +80,20 @@ static bool ContainsCheckPrefixes(string str, string[] checkPrefixes)
             return regex.Count(str) > 0;
         }
 
+        /// <summary>
+        /// Verifies LLVM "<prefix>" directives, such as "<prefix>:", "<prefix>-LABEL:", etc.. are valid.
+        /// Currently only checks to see if the user is using '-NEXT-FULL-LINE:' instead of '-FULL-LINE-NEXT:'.
+        /// </summary>
+        static void VerifyCheckPrefixes(string str, string[] checkPrefixes)
+        {
+            var invalidFullLinePattern = $"({String.Join('|', checkPrefixes)})+?({{LITERAL}})?(-NEXT-FULL-LINE:)";
+            var invalidRegex = new System.Text.RegularExpressions.Regex(invalidFullLinePattern);
+            if (invalidRegex.Count(str) > 0)
+            {
+                throw new SuperFileCheckException("'NEXT-FULL-LINE' is an invalid directive. Use 'FULL-LINE-NEXT'.");
+            }
+        }
```

Comment:
Is it possible to have a whitelist instead of blacklist for this?

Reply:
Given that we really just want to check for `-NEXT-FULL-LINE:` and because this syntax may change in the future, I don't think we really need to do more work here.

Reply:
Well, the point is to catch unrecognized directives that are noops, not just this one particular case.

Reply:
I guess ideally filecheck itself would warn for them though

Reply:
That makes sense. I do think this is fine for now though. I'm a little hesitant to try to do a catch-all for the directives considering the possibilities of the FileCheck syntax in general.

Reply:
Added an issue regarding it: https://github.com/dotnet/runtime/issues/82139